lib/OpenQA/Script/CloneJob.pm

```diff
@@ -245,8 +246,9 @@ sub clone_job ($jobid, $url_handler, $options, $post_params = {}, $jobs = {}, $d
     return $post_params if defined $post_params->{$jobid};
 
     my $job = $jobs->{$jobid} = clone_job_get_job($jobid, $url_handler, $options);
-    my $settings = $post_params->{$jobid} = {%{$job->{settings}}};
 
+    my $settings = $post_params->{$jobid} = {%{$job->{settings}}};
+    OpenQA::Script::CloneJobSUSE::detect_maintenance_update($jobid, $url_handler, $settings);
```

Comment:
I thought it was discussed that we don't want SUSE-specific code and we had a more generic proposal already 🤔

Reply:
See: https://github.com/os-autoinst/openQA/pull/4693#discussion_r920283603 (save @mimi1vx's comment on which variables to use)

Reply:
Not sure how that relates to my question. I was asking why we're not solving this in a generic way. I recall a conversation about a concrete implementation. Maybe @Martchus or @okurz knows

Reply:
@kalikiana can you explain better what generic way you're talking about? comment above states: The code works for osd and o3, without need for any modifications.

Reply:
@foursixnine To be honest I don't see any agreement in this PR, just a period of silence between the generic proposal and what's now implemented. If I was to guess there must have beem an out of band conversation that I was not part of.

> on one hand it's only checking for special test variables which shouldn't bother anyone not using those variables and on the other hand it can serve as a starting point to implement nicer plugins

I agree it doesn't hurt. I don't see the use case for nice plugins, though, if we think we can fully replace it with a generic solution. So if others are fine with it, fine, to me it seems like we'll duplicate the effort and have additional migrations following this.

Reply:
Btw just because I don't understand the rationale doesn't mean it's a bad idea. I'm just explaining my perspective here because I was surprised.

Reply:
> do you agree that we accept this PR after the smaller discussions are resolved?

Ok. I've also seen that most of my review comments have already been resolved. I'll re-review the PR later.